src/views/SessionView/index.vue

```diff
@@ -133,23 +135,42 @@ export default {
       promise = SessionService.newSession(
         this,
         type,
-        this.$route.params.subTopic
+        this.$route.params.subTopic,
+        {
+          onRetry: (res, abort) => {
+            this.showTroubleStartingModal(abort);
+          }
+        }
       );
       this.isNewSession = true;
     } else {
-      promise = SessionService.useExistingSession(this, id);
+      promise = SessionService.useExistingSession(this, id, {
+        onRetry: (res, abort) => {
+          this.showTroubleJoiningModal(abort);
+        }
+      });
       this.isNewSession = false;
     }
 
     promise
       .then(sessionId => {
-        this.$socket.connect();
-        this.joinSession(sessionId);
+        // ensure we restore user when we get a successful response
+        if (!this.isAuthenticated) {
+          this.$store.dispatch("user/fetchUser");
```

Comment:
@dmnisson could you clarify when this would be necessary?

Reply:
If someone tries to join a session from the dashboard with a broken connection, the call to `/api/user` will fail when the view changes and the user object will become empty. This ensures that the user object is restored when the connection is back online.